packages/searchkit-client/src/withSearchkitRouting.tsx

```diff
@@ -78,9 +88,12 @@ export default function withSearchkitRouting(
       const router = getRouting()
       if (router) {
         const routeState: RouteState = stateToRoute(searchkitVariables)
-        const currentRouteState = router.read()
+        const currentRouteState = {
+          size: api.baseSearchState.page?.size,
```

Comment:
the key to my change - before the routeState wasn't taking into account the size and therefore always adding a history entry even though the route state didn't actually change. Here i pull from the baseSearchState. Its overriden if there is one specified.